### PR TITLE
refactor: Harmonize fetching data from server

### PR DESCRIPTION
**Pull request by thielpa**
This PR changes the OpenApi generator library on the frontend to '@hey-api/openapi-ts'. This library directly generates Options for tanstack-query instead of angular services. Thus it removes some overhead.
Also all of the queries to the server use now tanstack queries.

PLEASE review and test this PR thoroughly because there are some critical changes to the codebase. I did my best in testing, but there might be some errors introduced.

Future:
Adding global error handling is not part of this PR, but should be done at some point.

### CODE REVIEW

**Comment by egekocabas** [comment]
@thielpa Do we need to change the openapi client code generation code in actions?

```
npx openapi-generator-cli generate -i ./server/application-server/openapi.yaml -g typescript-angular -o ./client/src/app/core/modules/openapi --additional-properties fileNaming=kebab-case,withInterfaces=true --generate-alias-as-model
```

**Comment by thielpa** [comment]
> @thielpa Do we need to change the openapi client code generation code in actions?
> 
> ```
> npx openapi-generator-cli generate -i ./server/application-server/openapi.yaml -g typescript-angular -o ./client/src/app/core/modules/openapi --additional-properties fileNaming=kebab-case,withInterfaces=true --generate-alias-as-model
> ```

Ahh yes we do. The client code is generated via a npm script now (`yarn generate:openapi`). All the config stuff is now in a config file

**Comment by egekocabas** [comment]
> > @thielpa Do we need to change the openapi client code generation code in actions?
> > ```
> > npx openapi-generator-cli generate -i ./server/application-server/openapi.yaml -g typescript-angular -o ./client/src/app/core/modules/openapi --additional-properties fileNaming=kebab-case,withInterfaces=true --generate-alias-as-model
> > ```
> 
> Ahh yes we do. The client code is generated via a npm script now (`yarn generate:openapi`). All the config stuff is now in a config file

Ok let me change that in this PR 👍🏻

File: client/src/app/pages/environment-deployment-history/environment-deployment-history.component.html [hunk 1]
```diff
@@ -17,21 +17,15 @@
...
-  <!-- Error State -->
-  @if (!isLoading() && isError()) {
+  } @else if (deploymentsQuery.isError()) {
+    <!-- Error State -->
     <div class="text-red-500">Error occurred while fetching deployments.</div>
-  }
-
-  <!-- Empty State -->
-  @if (isEmpty()) {
+  } @else if (deploymentsQuery.data.length === 0) {
```
Comment (egekocabas) [inline client/src/app/pages/environment-deployment-history/environment-deployment-history.component.html:23]:
```suggestion
  } @else if (deploymentsQuery.data()?.length === 0) {
```

File: client/src/app/pages/environment-deployment-history/environment-deployment-history.component.ts [hunk 2]
```diff
@@ -16,74 +14,17 @@ import { DateService } from '@app/core/services/date.service';
...
-    this.route.paramMap.subscribe(paramMap => {
-      const id = Number(paramMap.get('id'));
-      this.environmentId.set(!isNaN(id) ? id : null);
-    });
-  }
+  environmentId = input.required({ transform: numberAttribute });
 
   // TODO: Fix the query to fetch deployments by environment ID
   // Switching from one history page to another causes this query not called for the new environment ID
   // Thus we see the previous environment's deployment history
```
Comment (egekocabas) [inline client/src/app/pages/environment-deployment-history/environment-deployment-history.component.ts:27]:
```suggestion
```

This is fixed by this PR 👍🏻

File: client/src/app/components/forms/environment-edit-form/environment-edit-form.component.ts [hunk 3]
```diff
@@ -17,56 +16,63 @@ import { ActivatedRoute, Router } from '@angular/router';
...
-    description: '',
-    installedApps: [] as string[],
-  });
+  constructor(private messageService: MessageService) {
+    // This effect is needed, because the form is initialized before the data is fetched.
+    // As soon as the data is fetched, the form is updated with the fetched data.
+    effect(() => {
+      if (this.environment()) {
+        this.environmentForm.patchValue(this.environment() || {});
+        console.log(this.environment());
```
Comment (egekocabas) [inline client/src/app/components/forms/environment-edit-form/environment-edit-form.component.ts:29]:
```suggestion
```

**Review by egekocabas — CHANGES_REQUESTED** [review]
I fixed the github actions & found couple of bugs but those are not related with this PR, i will create issues about them.

I only found one thing in the `environment-deployment-history` html. Other than, the fetching works pretty good, thanks 👍🏻

**Review by egekocabas — APPROVED** [review]
🎊

**Pull request merged**